Build fails with XCode 10, which removed libstdc++ headers

**To reproduce:**
1. Checkout Sourcepawn on a Mac machine with XCode 10+
1. Run `configure.py`
1. Run `ambuild` in `obj-mac-x86`

**Expected:**
Build finishes without error

**Actual:**
https://gist.github.com/maximsmol/8ce249ee24339be8e194ffd066fb59a9
```
error: include path for stdlibc++ headers not found; pass '-std=libc++' on the command line to use the libc++ standard library instead [-Werror,-Wstdlibcxx-not-found]
/Users/maximsmol/proj/source/sourcepawn/third_party/amtl/amtl/am-vector.h:33:10: fatal error: 'new' file not found
#include <new>
         ^~~~~
2 errors generated.
Build failed.
```

Note that Apple wrongly suggests setting `-std=libc++`, which is invalid for clang and will cause the build to fail.

**Possible fix:**
Transition to libc++ as default libc on Mac targets by changing `cxx.cflags` in `AMBuildScript`.

Add `'-stdlib=libc++'`, bump `-mmacosx-version-min` to `10.9`.

Need to bump `-mmacosx-version-min` or build fails with `error: invalid deployment target for -stdlib=libc++ (requires OS X 10.7 or later)`. Note that `10.7` is not sufficient as clang will warn that `warning: libstdc++ is deprecated; move to libc++ with a minimum deployment target of OS X 10.9`.

**Note:**
An equivalent issue has been fixed by me in Metamod:Source

--- a/AMBuildScript
+++ b/AMBuildScript
@@ -178,12 +178,21 @@ class Config(object):
           cxx.postlink += ['-lgcc_eh']
         cxx.postlink += ['-lpthread', '-lrt']
       elif builder.target.platform == 'mac':
-        cxx.cflags += [
-          '-mmacosx-version-min=10.5',
-        ]
-        cxx.linkflags += [
-          '-mmacosx-version-min=10.5',
-        ]
+        if cxx.version >= 'apple-clang-10.0':
+          cxx.cflags += [
+            '-stdlib=libc++',
+            '-mmacosx-version-min=10.9',
+          ]
+          cxx.linkflags += [
+            '-mmacosx-version-min=10.9',
+          ]
+        else:
+          cxx.cflags += [
+            '-mmacosx-version-min=10.5',
+          ]
+          cxx.linkflags += [
+            '-mmacosx-version-min=10.5',
+          ]
       elif builder.target.platform == 'windows':
         cxx.defines += ['WIN32', '_WINDOWS']
 

